### PR TITLE
REGR: pd.to_hdf(..., dropna=True) not dropping missing rows

### LINKED ISSUE #35719: REGR: pd.to_hdf(dropna=True) not dropping all nan rows
#### Location of the documentation

[this should provide the location of the documentation, e.g. "pandas.read_csv" or the URL of the documentation, e.g. "https://dev.pandas.io/docs/reference/api/pandas.read_csv.html"]

**Note**: You can check the latest versions of the docs on `master` [here](https://pandas.pydata.org/docs/dev/).

https://pandas.pydata.org/pandas-docs/stable/user_guide/io.html

#### Documentation problem

[this should provide a description of what documentation you believe needs to be fixed/improved]

On the section:

"HDFStore will by default not drop rows that are all missing. This behavior can be changed by setting dropna=True."

The last example:

In [370]: pd.read_hdf('file.h5', 'df_with_missing')
Out[370]: 
   col1  col2
0   0.0   1.0
1   NaN   NaN
2   2.0   NaN

should output:

   col1  col2
0   0.0   1.0
1   2.0   NaN

which does not have rows that are all missing.

#### Suggested fix for documentation

[this should explain the suggested fix and **why** it's better than the existing documentation]

This has been provided above.

### PR DESCRIPTION
**Pull request by arw2019**
- [x] closes #35719
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.1.5.rst [hunk 1]
```diff
@@ -14,8 +14,7 @@ including other versions of pandas.
 
 Fixed regressions
 ~~~~~~~~~~~~~~~~~
--
--
+- Fixed regression in :meth:`DataFrame.to_hdf` was not dropping missing rows with ``dropna=True`` (:issue:`35719`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.5.rst:17]: this has been quite a while, move this to 1.2

File: pandas/tests/io/pytables/test_store.py [hunk 2]
```diff
@@ -1253,17 +1253,32 @@ def test_append_all_nans(self, setup_path):
...
         with ensure_clean_path(setup_path) as path:
-            df_with_missing.to_hdf(path, "df_with_missing", format="table")
-            reloaded = read_hdf(path, "df_with_missing")
+            df_with_missing.to_hdf(path, "df", format="table", dropna=False)
+            reloaded = read_hdf(path, "df")
             tm.assert_frame_equal(df_with_missing, reloaded)
 
+        with ensure_clean_path(setup_path) as path:
+            df_with_missing.to_hdf(path, "df", format="table", dropna=True)
+            reloaded = read_hdf(path, "df")
```
Comment (jreback) [inline pandas/tests/io/pytables/test_store.py:1279]: hmm, do we recreate this exactly, I think so but don't really remember (is expected the same)?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/io/pytables/test_store.py (hunk 2 above)
Comment (arw2019) [reply pandas/tests/io/pytables/test_store.py:1279]: I cherry-picked the example so it roundtrips

there are some dtype issues, for example a `RangeIndex` will come back as `Int64Index`. I'll look in the issue tracker and open a ticket if there isn't one already

File: doc/source/whatsnew/v1.1.5.rst (hunk 1 above)
Comment (arw2019) [reply doc/source/whatsnew/v1.1.5.rst]: Done

**Comment by arw2019** [comment]
Ready for re-review (CI green + addressed comments)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @arw2019 very nice

**Pull request merged**